src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -732,4 +765,16 @@ CHIP_ERROR OTARequestor::SendNotifyUpdateAppliedRequest(OperationalDeviceProxy &
     return cluster.InvokeCommand(args, this, OnNotifyUpdateAppliedResponse, OnNotifyUpdateAppliedFailure);
 }
 
+// Invoked when the device becomes commissioned
+void OTARequestor::OnCommissioningCompleteRequestor(const DeviceLayer::ChipDeviceEvent * event, intptr_t arg)
+{
+    ChipLogProgress(SoftwareUpdate, "Device commissioned, schedule a default provider query");
+
+    // TODO: Should we also send UpdateApplied here?
+
+    // Schedule a query. At the end of this query/update process the Default Provider timer is started
+    OTARequestorDriver * driver = (reinterpret_cast<OTARequestor *>(arg))->mOtaRequestorDriver;
+    driver->OTACommissioningCallback();
```

Comment:
Shouldn't you check for `event->Type == DeviceLayer::DeviceEventType::kCommissioningComplete` here?

Reply:
Also, the intention of this callback is to kick off our periodic timer in the driver as soon as the device is commissioned. But does this callback get invoked even if the device is already commissioned on bootup?

Reply:
Fixed

Reply:
Not sure if the callback is invoked if the device is already commissioned on bootup.
This would need to be followed up in another PR.